moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -963,21 +964,33 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
   {
     moveit::core::RobotStatePtr waypoint = trajectory.getWayPointPtr(p);
     Eigen::VectorXd new_point(num_joints);
+
     // The first point should always be kept
-    bool diverse_point = (p == 0);
+    if (p == 0)
+    {
+      points.push_back(new_point);
+      continue;
+    }
 
+    bool diverse_point = false;
     for (size_t j = 0; j < num_joints; ++j)
     {
       new_point[j] = waypoint->getVariablePosition(idx[j]);
-      // If any joint angle is different, it's a unique waypoint
-      if (p > 0 && std::fabs(new_point[j] - points.back()[j]) > min_angle_change_)
+      // If any joint angle is different and duration is nonzero, it's a unique waypoint
```

Comment:
```suggestion
      // If any joint angle is different, it's a unique waypoint
```

durations are not handled here

Reply:
I don't know, please discuss this in #2957.
I will close this to not clutter the conversation.